raiwidgets/raiwidgets/interfaces.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) Microsoft Corporation
+# Licensed under the MIT License.
+
+from enum import Enum
+from typing import List
+
+
+class Dataset(object):
+    predictedY: List
+    features: List[List[int]]
+    featureNames: List[str]
+    probabilityY: List
+    trueY: List
+    classNames: List[str]
+
+
+class BoundedCoordinates(object):
+    type: str
+    names: List
+    scores: List
+    scores_range: List
+    upper_bounds: List
+    lower_bounds: List
+
+
+class EBMGlobalExplanation (object):
+    feature_list: List[BoundedCoordinates]
+
+
+class FeatureImportance(object):
+    scores: List
+    intercept: float
+    featureNames: List[str]
+
+
+class PrecomputedExplanations(object):
+    localFeatureImportance: FeatureImportance
+    globalFeatureImportance: FeatureImportance
+    ebmGlobalExplanation: EBMGlobalExplanation
+    customVis: str
+
+
+class ModelClass(str, Enum):
+    Tree = "Tree"
+    EBM = "EBM"
+    blackbox = "blackbox"
+
+
+class ModelMethod(str, Enum):
+    classifier = "classifier"
+    regressor = "regressor"
+
+
+class ModelExplanationData(object):
```

Comment:
do we need to duplicate all of these compared to error analysis?

Reply:
same as previous, I don't want to bother to share things with assumption error-analysis and interpret dashboard will be gone eventually, and the code even right now are different between error-analysis and mad.

And this one is done through contract(class) vs error-analysis and interpret was defined by constant of property name.

So this way has extra benefit on define nested structure.
Eventually I will find someway either convert py to ts or the other way, so we can make sure contract synced between backend and front end. To do that I will have to create this python class contract first and use it in py.